fitsio/fitsio_pywrap.c

```diff
@@ -4016,21 +4022,46 @@ PyFITSObject_read_header(struct PyFITSObject* self, PyObject* args) {
         // this just returns the character string stored in the header; we
         // can eval in python
         if (fits_read_keyn(self->fits, i+1, keyname, value, comment, &status)) {
-            // is this enough?
             Py_XDECREF(list);
             set_ioerr_string_from_status(status);
             return NULL;
         }
+        ls=strlen(keyname);
+        tocomp = (ls < lcont) ? ls : lcont;
+        if (strncmp(keyname,"CONTINUE",tocomp)==0) {
+            continue;
+        }
+
+        if (fits_read_key_longstr(self->fits, keyname, &longstr, comment, &status)) {
+            Py_XDECREF(list);
+            set_ioerr_string_from_status(status);
+            return NULL;
+        }
+
+        if (card[8]=='=' && card[10]=='\'') {
+            is_string_value=1;
+        } else {
+            is_string_value=0;
+        }
+
 
         dict = PyDict_New();
-        add_string_to_dict(dict,"card_string",card);
         add_string_to_dict(dict,"name",keyname);
-        add_string_to_dict(dict,"value",value);
         add_string_to_dict(dict,"comment",comment);
+        add_long_to_dict(dict, "is_string_value", is_string_value);
+
+        // if not a comment but empty value, put in None
+        tocomp = (ls < lcomm) ? ls : lcomm;
+        if (0==strlen(longstr) && strncmp(keyname,"COMMENT",tocomp)!=0) {
+            add_none_to_dict(dict, "value");
+        } else {
+            add_string_to_dict(dict,"value",longstr);
+        }
+
+        free(longstr);
```

Comment:
I should set `longstr=NULL` after the free

Reply:
I didn't look at the C. :p